### PR TITLE
imaplib: fix bytes usage

### PR DESCRIPTION
**Pull request by sobolevn**
- `send` calls `_socket.sendall`
- `append` calls `MapCRLF.sub(CRLF, message)`

```python
>>> import imaplib
>>> imaplib.ParseFlags(bytearray())
()
>>> imaplib.Internaldate2tuple(bytearray())
>>> 
```

Source: https://github.com/python/cpython/blob/7ee3aca00a0dffc9cb5906a24005a8c61d5bce14/Lib/imaplib.py
Refs https://github.com/python/typeshed/issues/9006

### CODE REVIEW

File: stdlib/imaplib.pyi [hunk 1]
```diff
@@ -157,7 +157,7 @@ class _Authenticator:
     def encode(self, inp: bytes) -> str: ...
     def decode(self, inp: str) -> bytes: ...
 
-def Internaldate2tuple(resp: bytes) -> time.struct_time: ...
+def Internaldate2tuple(resp: ReadableBuffer) -> time.struct_time | None: ...
 def Int2AP(num: int) -> str: ...
```
Comment (JelleZijlstra) [inline stdlib/imaplib.pyi:161]:
```suggestion
def Int2AP(num: int) -> bytes: ...
```
Not directly related but this one definitely return bytes. We could also change the arg type to `SupportsAbs` I think.

Comment (sobolevn) [reply stdlib/imaplib.pyi]: Yes, sure! It is `num = int(abs(num))`: `SupportsAbs[SupportsInt]` 😱

Comment (JelleZijlstra) [reply stdlib/imaplib.pyi]: Still got to fix the return type :)

**Review by JelleZijlstra — APPROVED** [review]

File: stdlib/imaplib.pyi (hunk 1 above)
Comment (AlexWaygood) [reply stdlib/imaplib.pyi]: > Yes, sure! It is `num = int(abs(num))`: `SupportsAbs[SupportsInt]` 😱

Could be worse:

https://github.com/python/typeshed/blob/b5c4580d529c03efe7b4546ed9d3632befd5fb49/stdlib/unittest/case.pyi#L219

**Pull request merged**